codegen.js

```diff
@@ -0,0 +1,16 @@
+module.exports = {
+  schema: "https://graphql-pokemon2.vercel.app/",
+  documents: ["./**/*.graphql"],
+  generates: {
+    "./generated/types.ts": {
+      plugins: [
+        "typescript",
+        "typescript-operations",
+        "typescript-react-apollo",
```

Comment:
These plugins are what:
- Create validated typescript types for all our queries/fragments/mutations, captured in the glob in line 3
- Generate the hooks (which aren't used in this PR, but will be soon.